keepalived/core/keepalived_netlink.c

```diff
@@ -1127,7 +1127,10 @@ netlink_if_address_filter(__attribute__((unused)) struct sockaddr_nl *snl, struc
 							 vrrp->ifp) &&
 						 vrrp->family == ifa->ifa_family &&
 						 vrrp->saddr.ss_family != AF_UNSPEC &&
-						 (!vrrp->saddr_from_config || is_tracking_saddr)) {
+						 (!vrrp->saddr_from_config || is_tracking_saddr) &&
+						 (vrrp->family != AF_INET6 || (vrrp->family == AF_INET6 &&
```

Comment:
This code is not optimal, although a compiler will probably optimise it; however being not optimal makes it much harder to work out what it is doing.

The basic issue is (a || (^a && b)) is identical to (a || b)